src/stylize.ts

```diff
@@ -0,0 +1,74 @@
+import {Channel} from 'vega-lite/src/channel';
+import {Type} from 'vega-lite/src/type';
+
+import {QueryConfig} from './config';
+import {SpecQueryModel} from './model';
+// import {Property} from './property';
+import {EncodingQuery, ScaleQuery} from './query/encoding';
+import {Schema} from './schema';
+import {Dict} from './util';
+
+export function stylize(answerSet: SpecQueryModel[], schema: Schema, opt: QueryConfig): SpecQueryModel[] {
+
+  answerSet = answerSet.map(function(specM) {
+    let encQIndex: Dict<EncodingQuery> = {};
+
+    if (opt.smallBandSizeForHighCardinality || opt.smallBandSizeForFacet) {
+
+      [Channel.ROW, Channel.Y, Channel.COLUMN, Channel.X].forEach((channel) => {
+        encQIndex[channel] = specM.getEncodingQueryByChannel(channel);
+      });
+
+      const yEncQ = encQIndex[Channel.Y];
+      if (yEncQ !== undefined) {
+        if (encQIndex[Channel.ROW] ||
+            schema.cardinality(yEncQ) > 10) {
+
+          if (yEncQ.scale === undefined) {
+            yEncQ.scale = yEncQ.scale || {};
```

Comment:
Oh good call about that.  However, you still don't need `yEncQ.scale = yEncQ.scale || {};` when scale is already `undefined`.  

Because of that I couldn't guess that you think about this from reading (distracted).  Plus, there are no comment explaining and no test case.  

Can you add comment saying why we have to explicitly check for undefined?